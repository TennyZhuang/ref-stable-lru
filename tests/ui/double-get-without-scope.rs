use ref_stable_lru::LruCache;
use std::num::NonZeroUsize;

fn main() {
    let mut cache: LruCache<&'static str, String> = LruCache::new(NonZeroUsize::new(2).unwrap());

    let x = cache.get(&"a").unwrap().as_str();
    // Should failed here, since `x` already mutually borrowed cache.
    let y = cache.get(&"b").unwrap().as_str();
    let z = cache.get(&"c").unwrap().as_str();
    [x, y, z].join(" ");
}
