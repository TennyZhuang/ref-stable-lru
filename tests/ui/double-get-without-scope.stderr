error[E0499]: cannot borrow `cache` as mutable more than once at a time
  --> tests/ui/double-get-without-scope.rs:9:13
   |
7  |     let x = cache.get(&"a").unwrap().as_str();
   |             ----- first mutable borrow occurs here
8  |     // Should failed here, since `x` already mutually borrowed cache.
9  |     let y = cache.get(&"b").unwrap().as_str();
   |             ^^^^^ second mutable borrow occurs here
10 |     let z = cache.get(&"c").unwrap().as_str();
11 |     [x, y, z].join(" ");
   |      - first borrow later used here

error[E0499]: cannot borrow `cache` as mutable more than once at a time
  --> tests/ui/double-get-without-scope.rs:10:13
   |
7  |     let x = cache.get(&"a").unwrap().as_str();
   |             ----- first mutable borrow occurs here
...
10 |     let z = cache.get(&"c").unwrap().as_str();
   |             ^^^^^ second mutable borrow occurs here
11 |     [x, y, z].join(" ");
   |      - first borrow later used here
